### ISSUE TITLE
No connection possible with Passwordprotection

### ISSUE THREAD

**Issue by ChrisDuck87** [nokxs/iobroker-javascript-vs-code-extension#90]
I still have the problem that I can not connect with password protection. When I remove the protection from iobroker, the add-in works fine. 
When I add the password protection, I still get the following error message (after I enter the password):

"Could not connect to ioBroker. Check your '.iobroker-config.json' for wrong configuration: Error: Could not connect to 'https://192.168.2.195:8081/' after 10 seconds."

I did it several times, created new users for a crosstest, but nothing worked (except removing password protection)
The .iobroker-config.json looks like this:

{
  "ioBrokerUrl": "https://192.168.2.195",
  "socketIoPort": 8081,
  "scriptRoot": "/",
  "adminVersion": "admin5",
  "allowSelfSignedCertificate": true,
  "username": "admin"
}

I use the Admin V 5.3.8
Thanks a lot in advance - it is great what you are doing!

**Comment by ChrisDuck87** [comment]
![image](https://user-images.githubusercontent.com/88966869/182481572-b12882ca-4b2a-4e95-a39f-429f1b345f49.png)

This is the statusbar

**Comment by nokxs** [comment]
The config itself looks good and the status bar indicates a password protected iobroker instance is detectec, which is also good. 

If this status bar message is shown, an input field should appear automatically at the top:
![iobroker_password_input](https://user-images.githubusercontent.com/5887690/182932629-6b606ec7-f7cf-4fe6-b44a-c62182b7a3aa.png)

Does it not appear in your case? Have you tried restarting VS Code?

**Comment by ChrisDuck87** [comment]
Hey Nokxs, 

yes the status bar is displayed and after i put in my credentials and press "enter" the error occurs. 
i use special characters like # in my password. 

i have also done following tests: 

Delete the password from Iobroker: It worked fine
Create a new user with the password "Hallo1234" -> didnt worked

Of course i am available for a live test if needed.

**Comment by Sputnik24** [comment]
Doesn't work for me, neither.

Passwort prompt is shown, I enter my password and hit enter.

`Could not connect to ioBroker. Check your '.iobroker-config.json' for wrong configuration: Error: Could not connect to 'https://10.0.0.9:8081/' after 10 seconds.`

my config:
`{
  "ioBrokerUrl": "https://10.0.0.9",
  "socketIoPort": 8081,
  "scriptRoot": "/",
  "adminVersion": "admin5",
  "allowSelfSignedCertificate": true,
  "username": "admin"
}`

Using admin v5.3.8

**Comment by nokxs** [comment]
I was not lucky reproducing the problem. That is why I just released version 1.2.0, which adds the possibility to collect a `debug.log` file. The log collection is disabled by default and has to be manually enabled. Plase modify your config and add `"collectDebugLog": true`:

```json
{
  "ioBrokerUrl": "https://127.0.0.1",
  "socketIoPort": 8081,
  "scriptRoot": "scripts",
  "adminVersion": "admin5",
  "allowSelfSignedCertificate": true,
  "username": "admin",
  "collectDebugLog": true
}
```

Restart VS Code. From now on, a file `debug.log` should appear in the root of your wokspace. To stop the logging for this session you can execute the command `iobroker: Start/stop collecting debug log` over the command pallette (press F1). To permanently disable it, modify the config again.

Please try to reproduce your problem and send me the log file. Review it before and ensure no sensitive data is included.

**Comment by ChrisDuck87** [comment]
[debug.log](https://github.com/nokxs/iobroker-javascript-vs-code-extension/files/9277579/debug.log)

Hey Thanks a lot. 
Enclose you can find the debug log.

**Comment by nokxs** [comment]
Thanks for the log. From the log it looks like the login detection is not working as expected.

I've just released version 1.1.1, which includes improvements in the login process and has some more debug log output. Could check if its possible for you to login with this version? 

Could you send me the new `debug.log`, nevertheless if the log in works?

Thanks!

**Comment by ChrisDuck87** [comment]
[debug.log](https://github.com/nokxs/iobroker-javascript-vs-code-extension/files/9294762/debug.log)

I tried but i doesn't worked. Enclose you find the new debug.log including 1.1.1

**Comment by nokxs** [comment]
According to the log this is the problem: `unable to verify the first certificate`

It seems the certificate can not be verified, although the certificate should not be checked. It seems, it is not working your case. 

Are you using the certificate installed with ioBroker or have installed your own? According to StackOverflow it might be misconfigured certificate chain: https://stackoverflow.com/a/60020493

I will try this soon and supply you a version for testing: https://stackoverflow.com/a/32440021

**Comment by Sputnik24** [comment]
I also tried it, seems I have another error with self signed certificate (using the one generated by iobroker):
[debug.log](https://github.com/nokxs/iobroker-javascript-vs-code-extension/files/9305656/debug.log)

**Comment by nokxs** [comment]
Thanks for the logs. They are helping a lot!

I've tried something new, but I'm really not sure if its working. To try it, download the extension here: [iobroker-javascript-1.1.2-ssl-problems.1.vsix.zip](https://github.com/nokxs/iobroker-javascript-vs-code-extension/files/9329250/iobroker-javascript-1.1.2-ssl-problems.1.vsix.zip)

Unzip it and run this ommand in the directory you unzipped the vsix to: `code --install-extension iobroker-javascript-1.1.2-ssl-problems.1.vsix`
 
In my case I got a warning, but its working fine:
![grafik](https://user-images.githubusercontent.com/5887690/184425150-2f9c097e-36d2-4eca-9ab2-1183cf2dc5a2.png)

*What did I change?*
I added this line, which disables ssl certificate checks for the whole process.

```javascript
process.env['NODE_TLS_REJECT_UNAUTHORIZED'] = '0';
```

**Comment by ChrisDuck87** [comment]
Hey, thanks. i will try it these days. But just to be sure: 
I unizip it in e.g. C:\data and the command then in C:\data - correct?

**Comment by nokxs** [comment]
Yes, unzip it and then run the command. If everything worked fine, you should see the version in the extensions tab in vs code.

**Comment by knylbyte** [comment]
I also had problems with the default self signed certificate from ioBroker but with your fix everything is fine. Now I am able to connect to my ioBroker. Thanks

**Comment by ChrisDuck87** [comment]
sorry for the late response (due to illness)
i tried it too and it worked. Everything is fine and i have now access to the Scripts - Only point: I have to add  --force because it seems there is now a newer release in place:

![image](https://user-images.githubusercontent.com/88966869/191121901-dd261d14-9a6f-4a73-9c2d-47d30db0fdb2.png)


@nokxs: Thanks a lot!

Edit: Could it be that the version 1.1.2 which is in place override the fix? 

![image](https://user-images.githubusercontent.com/88966869/191124046-69427b48-5553-4978-a14b-8ff6fd26b805.png)

**Comment by chainnnnn** [comment]
Hi, I have the same issue (version 1.1.2 installed). Any news?

`2022-09-21T17:09:49.737Z ERROR Message Could not connect to ioBroker. Check your '.iobroker-config.json' for wrong configuration: Error: unable to verify the first certificate`

{
  "ioBrokerUrl": "https://192.168.178.102",
  "socketIoPort": 8081,
  "scriptRoot": "/",
  "adminVersion": "admin6",
  "allowSelfSignedCertificate": true,
  "username": "admin",
  "collectDebugLog": true
}

**Comment by nokxs** [comment]
This bug is fixed in 1.1.3

**Comment by RwithoutR** [comment]
Just for information if someone meets the same problem:
I could not connect to the ioBroker until i created a new admin user which i added then to the .iobroker-config.json.
The initial admin user did not work and i got the following error messages after entering the password.

ioBroker.javascript  v1.3.0 
ioBroker admin v6.12.0

**Visual Studio message**
Could not connect to ioBroker. Check your '.iobroker-config.json' for wrong configuration: Error: Could not connect to 'http://192.168.8.105:8081/' after 10 seconds.

**debug.log**
2023-11-04T10:57:13.005Z INFO SocketIoClient Try to connect to http://192.168.8.105:8081/
2023-11-04T10:57:13.006Z INFO SocketIoClient Created websocket with authorization. AllowSelfSignedCertificate: true. Cooke set: yes
2023-11-04T10:57:13.013Z INFO SocketIoClient type: message, data: [0,401,"reauthenticate"]
2023-11-04T10:57:16.014Z WARN SocketIoClient No READY flag received in 3 seconds. Re-init
